### PR TITLE
Add support for Zookeeper

### PR DESCRIPTION
**Pull request by dougszumski**

### CODE REVIEW

File: ansible/group_vars/all/kolla [hunk 1]
```diff
@@ -181,6 +181,8 @@ overcloud_container_image_regex_map:
     enabled: "{{ kolla_enable_swift | bool }}"
   - regex: tgtd
     enabled: "{{ kolla_enable_cinder | bool or kolla_enable_ironic | bool }}"
+  - regex: zookeeper
+    enabled: "{{ kolla_enable_zookeeper | bool or kolla_enable_kafka | bool }}"
```
Comment (markgoddard) [inline ansible/group_vars/all/kolla:185]: kolla_enable_kafka needs a default

File: ansible/roles/kolla-openstack/defaults/main.yml [hunk 2]
```diff
@@ -348,3 +348,12 @@ kolla_extra_sahara:
...
 kolla_enable_swift:
+
+###############################################################################
+# Zookeeper configuration.
+
+# Whether to enable Zookeeper.
+kolla_enable_zookeeper:
+
+# Free form extra configuration to append to zookeeper.cfg.
+kolla_extra_zookeeper:
```
Comment (markgoddard) [inline ansible/roles/kolla-openstack/defaults/main.yml:359]: Need to add these to the enable-everything molecule scenario for kolla-openstack

**Review by markgoddard — CHANGES_REQUESTED** [review]
Has a kolla-ansible zk role been developed locally?

File: ansible/roles/kolla-openstack/vars/main.yml [hunk 3]
```diff
@@ -108,3 +108,8 @@ kolla_openstack_custom_config:
     dest: "{{ kolla_node_custom_config_path }}/swift"
     patterns: "*"
     enabled: "{{ kolla_enable_swift }}"
+  # Zookeeper.
+  - src: "{{ kolla_extra_config_path }}/zookeeper"
+    dest: "{{ kolla_node_custom_config_path }}/zookeeper"
+    patterns: "*"
+    enabled: "{{ kolla_enable_zookeeper }}"
```
Comment (markgoddard) [inline ansible/roles/kolla-openstack/vars/main.yml:115]: Needs adding to the list in ansible/roles/kolla-openstack/molecule/default/tests/test_default.py and ansible/roles/kolla-openstack/molecule/enable-everything/tests/test_default.py.

**Review by markgoddard — CHANGES_REQUESTED** [review]

**Comment by dougszumski** [comment]
Need to debug molecule tests...

**Comment by dougszumski** [comment]
Tests are fixed.

Note to self: squash before merge.

**Review by markgoddard — APPROVED** [review]
Looks good. Thanks for updating the tests.

**Comment by dougszumski** [comment]
Re-submitted via OpenStack infra: https://review.openstack.org/#/c/552622/

**Pull request closed without merging**